lib/lambda_ethereum_consensus/store/blocks.ex

```diff
@@ -0,0 +1,68 @@
+defmodule LambdaEthereumConsensus.Store.Blocks do
+  @moduledoc false
+  alias LambdaEthereumConsensus.Store.BlockStore
+
+  use GenServer
+
+  @ets_block_by_hash __MODULE__
+
+  ##########################
+  ### Public API
+  ##########################
+
+  def start_link(_opts) do
+    GenServer.start_link(__MODULE__, [], name: __MODULE__)
+  end
+
+  def store_block(block_root, signed_block) do
+    cache_block(block_root, signed_block)
+    GenServer.cast(__MODULE__, {:store_block, block_root, signed_block})
+  end
+
+  def get_block(block_root), do: lookup(block_root)
+
+  @spec clear() :: any()
+  def clear, do: :ets.delete_all_objects(@ets_block_by_hash)
+
+  ##########################
+  ### GenServer Callbacks
+  ##########################
+
+  @impl GenServer
+  def init(_) do
+    :ets.new(@ets_block_by_hash, [:set, :public, :named_table])
+    {:ok, %{}}
+  end
+
+  @impl GenServer
+  def handle_cast({:store_block, block_root, signed_block}, state) do
+    BlockStore.store_block(signed_block, block_root)
+    # TODO: remove old blocks from cache
+    {:noreply, state}
+  end
+
+  ##########################
+  ### Private Functions
+  ##########################
+
+  defp lookup(block_root) do
+    with nil <- :ets.lookup_element(@ets_block_by_hash, block_root, 2, nil),
+         block when not is_nil(block) <- fetch_block(block_root) do
+      cache_block(block_root, block)
+      block
+    end
```

Comment:
This comment is optional and subjective, but I think this is one of those cases where the `with` is less clear than a simple case. Maybe something like this is better?

```elixir
case :ets.lookup_element(@ets_block_by_hash, block_root, 2, nil) do
  nil -> cache_miss(block_root)
  block -> block
end
```

Then `cache_miss` itself can be the one that performs both the lookup and add it to the store.

Reply:
Done